### ISSUE TITLE
Reduce Dependence on malloc()

### ISSUE THREAD

**Issue by hyc** [openssl/openssl#18317]
Now that OpenSSL 3.0 has made most structures opaque and required a _new() function to construct them, a lot of code that used to have no malloc dependencies is now quite malloc heavy. This effect could be mitigated if the APIs provided _size() functions that returned the sizeof() their respective internal structures, to allow use of alloca(). E.g.:
```
size_t EVP_MD_CTX_size();
#define EVP_MD_CTX_LOCAL()  alloca(EVP_MD_CTX_size())
```

This should typically be as safe in terms of stack consumption as the old code that just declared CTXs as local variables.

**Comment by paulidale** [comment]
I rather suspect that this won't help as much as you imagine.

**Comment by nhorman** [comment]
Marking as inactive, to be closed when 3.4 dev cycle ends, barring further input

**Comment by hyc** [comment]
I will provide benchmark/profiling results showing the difference.
